File: gdbm/jython/update/gdbm_update.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	gdbm_update.py
#
#					Jul/09/2010
import	sys
import	string
import anydbm
#
sys.path.append ('/var/www/uchida/data_base/common/python_common')
#
from dbm_manipulate import dbm_disp_proc,dbm_update_proc
# -------------------------------------------------------------
print ("*** 開始 ***")
#
id_in = string.atoi (sys.argv[1])
population_in = string.atoi (sys.argv[2])
print ("%d\t%d" % (id_in, population_in))
#
db_name = "/var/tmp/gdbm/cities.pag";
dd = anydbm.open (db_name,"c")
dbm_update_proc	(dd,id_in,population_in)
dbm_disp_proc (dd)
#
dd.close ()
#
print ("*** 終了 ***")
# -------------------------------------------------------------

File: gdbm/jruby/update/gdbm_update.rb
#! /usr/bin/ruby
# -*- encoding: utf-8 -*-
#
#	update/gdbm_update.rb
#
#				Aug/18/2010
#
require	'dbm'
require	'date'
require	'json'
#
load '/var/www/uchida/data_base/common/ruby_common/kvalue_manipulate.rb'
# --------------------------------------------------------------------
puts	"*** 開始 ***"
#
id_in = ARGV[0].to_i
population_in = ARGV[1].to_i
#
puts id_in,population_in
#
ee=DBM.open("/var/tmp/gdbm/cities")
#
kvalue_update_proc(ee,id_in,population_in)
#
ww=ee.to_hash;
kvalue_display_proc(ww)
ee.close
#
print "ww.size =",ww.size,"\n"
#
puts "*** 終了 ***"
#
